### ISSUE TITLE
Please send pull requests to Crystal :-)

### ISSUE THREAD

**Issue by asterite** [datanoise/openssl.cr#1]
Hi!

The code you are writing in Crystal is quite amazing! Mongo, AMQP, and now correct and complete bindings for OpenSSL.

The OpenSSL bindings in Crystal are poor/incomplete because we didn't have time to finish them, and we also lack a total understanding of its API (which is, again, just lack of time). If you can, please submit pull requests with your additions/changes, we would be more than happy to merge them :-)

I also really like that you wrote specs for this.

Our idea of Crystal's standard library is that it has support for most of the standard libraries and protocols out there. OpenSSL is very common so it would be nice to have excellent support for it out of the box.

**Comment by datanoise** [comment]
Yes, my plan was to create a pull request eventually. At this point I'd like to have a separate repository as a playground where I can add features/tests without constantly making pull requests. But definitely I do like to see it in a standard library. Thanks for the kind words!

**Comment by bararchy** [comment]
+1 for PR to Crystal, I've been playing around with adding missing things to the OpenSSL lib also, you can see here: https://github.com/manastech/crystal/pull/669

If you want I can also help you test and open issues until you feel your lib is stable enough to be added into the standard library.

**Comment by datanoise** [comment]
@bararchy absolutely. any help is appreciated.

**Comment by asterite** [comment]
@datanoise @bararchy Of course, take your time, there's absolutely no rush in any of this. And big thanks for the help! :-)
